src/components/AddTechDialog.tsx

```diff
@@ -0,0 +1,177 @@
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { makeStyles, Theme, createStyles, FormControl, Grid } from '@material-ui/core';
+import { useSnackbar } from 'notistack';
+import { addTech } from '../api/techs/addTech';
+import { User } from '../models/user';
+import { TextField } from '@material-ui/core';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    form: {     
+      display: "flex",
+      flexDirection: "column",
+      alignItems: "center",
+    },
+    addButton: {
+      margin: theme.spacing(2),
+    },
+    formControl: {
+        minWidth: 160,
+    },
+  })
+);
+
+export interface AddTechDialogProps {
+  setTechs: (value: React.SetStateAction<User[]>) => void;
+}
+
+const AddTechDialog = (props: AddTechDialogProps) => {
+  const classes = useStyles();
+  const [open, setOpen] = React.useState(false);
+
+  const [username, setUsername] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [confirmPassword, setConfirmPassword] = useState<string>('');
+
+  const [submitted, setSubmitted] = useState<boolean>(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+    setSubmitted(false);
+  }
+
+  const handleClickClose = () => {
+    setOpen(false);
+
+    setUsername('');
+    setPassword('');
+    setConfirmPassword('');
+    setSubmitted(false);
+  }
+
+  const { enqueueSnackbar } = useSnackbar();
+
+  const handleUsernameChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setUsername(event.target.value as string);
+  };
+
+  const errorUsername = () => {
+    return username === '';
+  }
+
+  const handlePasswordChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setPassword(event.target.value as string);
+  };
+
+  const errorPassword = () => {
+    console.log("Password: " + password)
+    return password === '';
+  }
+
+  const handleConfirmPasswordChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setConfirmPassword(event.target.value as string);
+  };
+
+  const errorConfirmPassword = () => {
+    return confirmPassword === '' || password !== confirmPassword;
+  }
+
+  const errorForm = () => {
+    return errorUsername() || errorPassword() || errorConfirmPassword();
+  }
+
+  const handleSubmit = () => {
+    setSubmitted(true);
+    if (errorForm()) return;
+    const tech = addTech(username, password);
+    tech.then(response => {
+        if(response.isError)
+        {
+          enqueueSnackbar("Could not add tech", { variant: "error" });
+        }
+        else if(response.data)
+        {
+          props.setTechs(prev => 
+            response.data ? 
+            prev = [...prev, response.data]
+            :          
+            prev
```

Comment:
I think it looks better in one line

Reply:
it looks even beter without `?` becouse `response.data` is true (you are checking it in else if above)

Reply:
also never assign values to prev

Reply:
```suggestion
        const data = response.data;
        props.setTechs(prev => [...prev, data])
```

Reply:
Fixed it just like you suggested